scipy/sparse/lil.py

```diff
@@ -450,18 +450,31 @@ def tolil(self, copy=False):
     tolil.__doc__ = spmatrix.tolil.__doc__
 
     def tocsr(self, copy=False):
-        lst = [len(x) for x in self.rows]
-        idx_dtype = get_index_dtype(maxval=max(self.shape[1], sum(lst)))
-        indptr = np.cumsum([0] + lst, dtype=idx_dtype)
-        indices = np.empty(indptr[-1], dtype=idx_dtype)
-        data = np.empty(indptr[-1], dtype=self.dtype)
-
-        start = 0
-        for stop, indices_i, data_i in zip(indptr[1:], self.rows, self.data):
-            indices[start:stop] = indices_i
-            data[start:stop] = data_i
-            start = stop
-
+        # construct indptr array
+        M, N = self.shape
+        lengths = np.fromiter(map(len, self.rows), dtype=get_index_dtype(maxval=N), count=M)
```

Comment:
This line and others in this change should wrap at 80 columns.

Reply:
Updated accordingly.